core/frontend/src/render/webgl/ShaderBuilder.ts

```diff
@@ -1060,6 +1063,11 @@ export class FragmentShaderBuilder extends ShaderBuilder {
       main.addline("  assignFragData(baseColor);");
     }
 
+    if (this.requiresEarlyZWorkaround) {
+      // Add a conditional discard that never executes to force buggy Intel driver to skip early Z despite our fragment shader writing depth.
+      main.addline("if (v_eyeSpace.z == 9999999.0) discard;");
```

Comment:
@DStradley @MarcNeely please review this conditional. I didn't want to use something that could get optimized out so used a varying instead of a uniform. I am not sure if something simpler like `if (v_eyeSpace.z != v_eyeSpace.z)` would get optimized out by a smart compiler. v_eyeSpace is a varying that's present in all of the shaders that need patching, and eye Z should always be <= 0. Do you spot any problems or a better way to implement this?

Reply:
I think that should do it.  I would hope that you're right about it probably optimizing out the other.  You could always do a "dta output shaders c" to compile and output all of the shaders and then run the _makeShade.bat file to generate the HLSL for it and then check to make sure that it still has the discard in it (in the resulting .h file(s) for the shaders that should have it).